flake.nix

```diff
@@ -2,115 +2,123 @@
   description = "Compositor for the COSMIC desktop environment";
 
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    # <https://github.com/nix-systems/default-linux>
+    systems.url = "github:nix-systems/default-linux";
 
-    parts.url = "github:hercules-ci/flake-parts";
-    parts.inputs.nixpkgs-lib.follows = "nixpkgs";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
 
     crane.url = "github:ipetkov/crane";
 
-    rust.url = "github:oxalica/rust-overlay";
-    rust.inputs.nixpkgs.follows = "nixpkgs";
-
-    nix-filter.url = "github:numtide/nix-filter";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    rust-overlay.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  outputs =
-    inputs@{
-      self,
-      nixpkgs,
-      parts,
-      crane,
-      rust,
-      nix-filter,
-      ...
-    }:
-    parts.lib.mkFlake { inherit inputs; } {
-      systems = [
-        "aarch64-linux"
-        "x86_64-linux"
-      ];
-
-      perSystem =
-        {
-          self',
-          lib,
-          system,
-          ...
-        }:
-        let
-          pkgs = nixpkgs.legacyPackages.${system}.extend rust.overlays.default;
-          rust-toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
-          craneLib = (crane.mkLib pkgs).overrideToolchain rust-toolchain;
-          craneArgs = {
-            pname = "cosmic-comp";
-            version = self.rev or "dirty";
-
-            src = nix-filter.lib.filter {
-              root = ./.;
-              include = [
-                ./src
-                ./i18n.toml
-                ./Cargo.toml
-                ./Cargo.lock
-                ./resources
-                ./cosmic-comp-config
-              ];
-            };
-
-            nativeBuildInputs = with pkgs; [
-              pkg-config
-              autoPatchelfHook
-              cmake
-            ];
+  outputs = {
+    self,
+    crane,
+    nixpkgs,
+    rust-overlay,
+    systems,
+    ...
+  }: let
+    inherit (nixpkgs) lib;
 
-            buildInputs = with pkgs; [
-              wayland
-              systemd # For libudev
-              seatd # For libseat
-              libxkbcommon
-              libinput
-              mesa # For libgbm
-              fontconfig
-              stdenv.cc.cc.lib
-              pixman
-              libdisplay-info
-            ];
+    fs = lib.fileset;
+    eachSystem = lib.genAttrs (import systems);
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
+        localSystem.system = system;
+        overlays = [(import rust-overlay)];
+      });
+  in {
+    packages =
+      lib.mapAttrs (system: pkgs: let
```

Comment:
Optional, can be changed later: 

If you wish to provide an overlay, for users that want to introduce `cosmic-comp` into `pkgs`, and build it with *their* nixpkgs instead of this flake's transitive input, move the package to a file (`nix/package.nix`) and make a flake output `overlays.default = final: prev: { cosmic-comp = final.callPackage "${self}/nix/package.nix" {}; };`, then `inherit (pkgsFor.${system}) cosmic-comp; default = self.packages.${system}.cosmic-comp;` in flake output `packages`.

Typing on a phone, sorry for errors.